p2p/protocols/protocol.go

```diff
@@ -359,7 +286,7 @@ func (p *Peer) Send(ctx context.Context, msg interface{}) error {
 
 	code, found := p.spec.GetCode(msg)
 	if !found {
-		return errorf(ErrInvalidMsgType, "%v", code)
+		return fmt.Errorf("invalid message type %v ", code)
```

Comment:
probably better to break here. if the message code is not found - it means the nodes are incompatible and we should disconnect from that node

Reply:
This is a sync call, as it is not using an event loop, so here we don't need to use a special error. This goes for `Handshake()` calls as well. Only run needs a special `QueryError` in order to break. Does this makes sense?